test/PushNotificationTests/PackagedTests.cpp

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include <TestDef.h>
+#include "MockBackgroundTaskInstance.h"
+#include <TlHelp32.h>
+#include "TestFunctions.h"
+
+using namespace WEX::Common;
+using namespace WEX::Logging;
+using namespace WEX::TestExecution;
+
+using namespace winrt::Windows::ApplicationModel::Activation;
+using namespace winrt::Windows::ApplicationModel::Background;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::Management::Deployment;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::System;
+using namespace winrt::Microsoft::Windows::PushNotifications;
+
+namespace Test::PushNotifications
+{
+    class PackagedTests
+    {
+
+    public:
+        BEGIN_TEST_CLASS(PackagedTests)
+            TEST_CLASS_PROPERTY(L"ThreadingModel", L"MTA")
+            TEST_CLASS_PROPERTY(L"RunFixtureAs:Class", L"RestrictedUser")
+            TEST_CLASS_PROPERTY(L"RunAs", L"UAP")
+            TEST_CLASS_PROPERTY(L"UAP:AppxManifest", L"PushNotifications-AppxManifest.xml")
+            TEST_CLASS_PROPERTY(L"UAP:Host", L"PackagedCWA")
+        END_TEST_CLASS()
+
+        TEST_CLASS_SETUP(ClassInit)
+        {
+            ::Test::Bootstrap::SetupPackages();
+            return true;
+        }
+
+        TEST_CLASS_CLEANUP(ClassUninit)
+        {
+            ::Test::Bootstrap::CleanupPackages();
+            return true;
+        }
+
+        TEST_METHOD_SETUP(MethodInit)
+        {
+            ::WindowsAppRuntime::SelfContained::TestInitialize(::Test::Bootstrap::TP::WindowsAppRuntimeFramework::c_PackageFamilyName);
```

Comment:
The namespace causes me confusion. This is a non-self-contained packaged app, but we are calling an API for self-contained apps?

Reply:
Similarly to MddBootstrapTestInitialize, SelfContained::TestInitialize sets a global string in the SDK used for testing. The implementation looks for a framework that is dynamically generated by the pipeline which will fail in inner loop testing. This API allows inner loop testing using the IsSelfContained API correcty.

Reply:
So, it isn't possible for packaged apps to make use of WinAppSDK as a non-self-contained app without this call?

Reply:
Yup :)